checks.d/network.py

```diff
@@ -440,3 +446,117 @@ def _parse_solaris_netstat(self, netstat_output):
             metrics_by_interface[iface] = metrics
 
         return metrics_by_interface
+
+    @staticmethod
+    def _linux_sockstat_get_conntrack_max():
+        proc_name = '/proc/sys/net/ipv4/ip_conntrack_max'
+        try:
+            proc = open(proc_name,'r')
+            try:
+                lines = proc.readline()
+            finally:
+                proc.close()
+            return int(lines)
+        except IOError:
+            return 0
+
+    @staticmethod
+    def _linux_sockstat_get_conntrack_count():
+        proc_name = '/proc/sys/net/ipv4/netfilter/ip_conntrack_count'
+        try:
+            proc = open(proc_name,'r')
+            try:
+                lines = proc.readline()
+            finally:
+                proc.close()
+            return int(lines)
+        except IOError:
+            return 0
+
+    @staticmethod
+    def _linux_sockstat_get_max_orphans(self):
+        proc_name = '/proc/sys/net/ipv4/tcp_max_orphans'
+        try:
+            proc = open(proc_name,'r')
+            try:
+                lines = proc.readline()
+            finally:
+                proc.close()
+            return int(lines)
+        except IOError:
+            self.log.debug("Unable to read %s." % proc_name)
+
+    @staticmethod
+    def _linux_sockstat_get_size_of_page():
+        return long(check_output(["/usr/bin/getconf", "PAGESIZE"]))
+
+    @staticmethod
+    def _linux_sockstat_get_max_memory(self):
+        sizeOfPageInBytes = Network._linux_sockstat_get_size_of_page()
+        tcpMemFilePath = '/proc/sys/net/ipv4/tcp_mem'
+        try:
+            proc = open(tcpMemFilePath,'r')
+            try:
+                tcpmem = proc.readlines()
+            finally:
+                proc.close()
+
+        except IOError:
+            self.log.debug("Unable to read %s." % tcpMemFilePath)
+        tcpmemline = tcpmem[0].split()
+        maxMemoryInBytes = sizeOfPageInBytes * tcpmemline[2]
+        return long(maxMemoryInBytes)
+
+    @staticmethod
+    def _linux_sockstat(self):
+        proc_name = '/proc/net/sockstat'
+        try:
+            proc = open(proc_name,'r')
+            try:
+                lines = proc.read()
+            finally:
+                proc.close()
+            reg = re.compile("sockets: used (?P<sockets_used>\d+)\n"
```

Comment:
Define your regex above the class in a variable such as `LINUX_SOCKSTATS_REGEX`. This makes it easier to manage all the metrics stuff for a given check.